x-pack/filebeat/input/awss3/config.go

```diff
@@ -33,6 +34,9 @@ type config struct {
 	AWSConfig           awscommon.ConfigAWS  `config:",inline"`
 	FileSelectors       []fileSelectorConfig `config:"file_selectors"`
 	ReaderConfig        readerConfig         `config:",inline"` // Reader options to apply when no file_selectors are used.
+	PathStyle           bool                 `config:"path_style"`
+	RegionOverride      string               `config:"region_override"`
+	ProviderOverride    string               `config:"provider_override"`
```

Comment:
Is there anyway we can infer the provider from the endpoint or any API request?
If we can avoid to specify it explicitly and let the code find it it would be better, what do you think?

Reply:
I think it would be hard unless you just Didi it by domain name but even then it would make `aws` into `amazonaws`